### PR TITLE
Expose double signer implementation

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

This PR provides double signer implementation, which is used to test the slashing feature. It relies on build tags to build honest and double signer implementations and removes byzantine binary from the repository.

doubleSigner build tag is introduced, so whenever we want to build a double signer implementation, we should run the following command:
`go build -tags doubleSigner -o artifacts/polygon-edge-double-signer .`

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: consensus/polybft/custom_multicast_double_signer.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"crypto/rand"
+
+	ibftProto "github.com/0xPolygon/go-ibft/messages/proto"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+func (p *Polybft) customMulticastHandler(msg *ibftProto.Message) {
+	if msg.Type == ibftProto.MessageType_COMMIT {
```
Comment (dusan-maksimovic) [inline consensus/polybft/custom_multicast_double_signer.go]: nitpick: maybe return early here if it is not commit message, and if sender != localAddr, to reduce nesting?

**Review by dusan-maksimovic — APPROVED** [review]
Nice & clean!

**Comment by Stefan-Ethernal** [comment]
> Nice & clean!

Credits to @vcastellm! :slightly_smiling_face:

File: consensus/polybft/custom_multicast_double_signer.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply consensus/polybft/custom_multicast_double_signer.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1875/commits/0543388e0ccaff0844f3f2d46debdaa512ba533c

**Review by jelacamarko — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]
🎩

**Review by goran-ethernal — APPROVED** [review]
Nice

**Review by stana-miric — APPROVED** [review]

**Pull request merged**